src/form-builder/components/FormBuilder.jsx

```diff
@@ -85,75 +151,165 @@ export default class FormBuilder extends Component {
-                  'resources:(value,dataType,uuid))';
-                httpInterceptor
-                  .get(`${formBuilderConstants.formUrl}/${formUuid}?${params}`)
-                  .then((data) => {
-                    const formResource = {
-                      form: {
-                        name: formName,
-                        uuid: formUuid,
-                      },
-                      value: JSON.stringify(value),
-                      uuid: data.resources[0].uuid,
-                    };
-
-                    self.props.saveFormResource(formResource, translations);
-                  });
-              });
-          });
+        const formData = JSON.parse(reader.result);
+        const formsValidationPromises = [];
+        const validateFormJsonPromise = self.validateFormJson(fileName, formData);
+        if (validateFormJsonPromise !== null) {
+          formsValidationPromises.push(validateFormJsonPromise);
         }
+        self.import(formsValidationPromises);
       } catch (error) {
-        self.props.onValidationError('Error Importing.. Please import a valid form');
+        self.updateImportErrors(fileName, self.parseErrorMessage);
+        self.downloadErrorsFile(self.importErrors);
       }
     };
-
     reader.readAsText(file[0]);
   }
 
+  validateFormJson(fileName, formData) {
+    const self = this;
+    const { formJson, translations } = formData;
+    const formName = formJson.name;
+    const value = JSON.parse(formJson.resources[0].value);
+    const form = {
+      name: formName,
+      version: '1',
+      published: false,
+    };
+    if (!formHelper.validateFormName(formName)) {
+      self.updateImportErrors(fileName, 'Invalid File Name');
+    } else {
+      const conceptValidationPromise = self.fixuuid(value, fileName);
+      return conceptValidationPromise.then(() => {
+        if (self.formConceptValidationResults[fileName]) {
+          const validationResults = self.formConceptValidationResults[fileName];
+          let message = 'Concept validation error: \n';
+          validationResults.forEach(v => {
+            message = message.concat(`${v}\n`);
+          });
+          self.updateImportErrors(fileName, message);
+        } else {
+          self.formJSONs.push({ form, value, formName, translations });
+        }
+      });
+    }
+    return null;
+  }
+
+  import(formsValidationPromises) {
+    const self = this;
+    Promise.all(formsValidationPromises).then(() => {
+      if (self.importErrors.length !== 0) {
+        self.downloadErrorsFile(self.importErrors);
+      }
+      if (self.formJSONs.length !== 0) {
+        self.importForms(self.formJSONs);
+      }
+    });
+  }
+
+  downloadErrorsFile(errors) {
+    const filename = 'importErrors.txt';
+    const blob = new Blob([JSON.stringify(errors)], {
+      type: 'text/plain',
+    });
+    saveAs(blob, filename);
+    this.hideLoader();
+  }
+
+  importForms(formJsons) {
```

Comment:
this method is the one actually making the api call to import. And i suggest it to rename the method to reflect the same as there are so many import** methods are there in the same component.